stats/Record.md

```diff
@@ -0,0 +1,71 @@
+# Record API Overview
+The stats library allows users to record metrics for their applications or libraries. The core 
+data types used are:
+* `Measure`: describes the type of the individual values recorded by an application.
+* `Measurement`: describes a data point to be collected for a `Measure`.
+
+## Measure
+A `Measure` describes the type of the individual values/measurements recorded by an application. It
+includes information such as the type of measurement, the units of measurement and descriptive names
+for the data. This provides the fundamental type used for recording data.
+
+A Measure describes a value with the following metadata:
+* `name`: a string by which the measure will be referred to, e.g. "rpc_server_latency", or
+"vm_cpu_cycles". Names MUST be unique within the library. It is recommended to use names 
+compatible with the intended end usage, e.g, use host/path pattern.
+* `description`: a string describes the measure, e.g. "RPC latency in seconds", "Virtual cycles
+executed on VM".
+* `unit`: a string describing the unit used for the `Measure`. Follows the format described by
```

Comment:
below unit referenced in value validation context, is it also exported somehow? what views know about units?

Reply:
Yes in the exported data we will include this. View do not use this unit but backends may use this to display in the UI or to apply data conversion (we record KB for each measurement but when displayed the backend may display MB/sec for example).